app/controllers/application_records_controller.rb

```diff
@@ -22,19 +22,27 @@ def create
   def csv_export
     start_date = parse_american_date(params.require :start_date)
     end_date = parse_american_date(params.require :end_date)
-    params.require :department_ids
-    @records = ApplicationRecord.in_department(params[:department_ids])
-               .between(start_date, end_date)
+    if params[:department_ids]
```

Comment:
Some pretty similar logic between the three controller actions here, so a `before_action` is probably warranted.